cards.js

```
@@ -0,0 +1,121 @@
+// constructor for card object
+function Card(name) {
+  this.name = !name ? 'generic card' : name;
+}
+
+// constructor for deck object
+function Deck(name, size) {
+  this.name = name;
+  this.size = size;
+  this.cards = [];
+  this.pile = [];
+  this.hand = [];
+}
+
+// builds deck with certain cards
+Deck.prototype.build = function() {
+  for (let i = 1; i <= this.size; i++) {
+    let card = new Card('card ' + i);
+    this.cards.push(card);
+  }
+};
+
+// shuffle deck
+Deck.prototype.shuffle = function() {
+  // copy the cards blueprint
+  let temp = this.cards.slice(0);
+
+  while (temp.length > 0) {
+    let index = getRandomInt(0, temp.length);
+    let randCard = temp[index];
+    this.pile.push(randCard);
+    temp.splice(index, 1);
+  }
+};
+
+// recursively draw 'num' amount of cards from the top of the deck
+Deck.prototype.draw = function(num) {
+  if (num === 0)
+    return;
```

If possible in javascript throwing an illegal argument exception would be best here instead of silently failing.